mathesar/api/db/viewsets/columns.py

```diff
@@ -34,7 +34,8 @@ def create(self, request, table_pk=None):
         # We only support adding a single column through the API.
         serializer = ColumnSerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
-
+        if request.data.get('type_options', None) is not None and request.data['type_options'].get('scale', None) is not None and request.data['type_options'].get('precision', None) is None:
```

Comment:
This is long and hard to parse quickly, can you put each conditional into a separate variable?